botfront/imports/ui/components/common/Browser.jsx

```diff
@@ -48,55 +74,98 @@ class Browser extends React.Component {
             allowAddition,
             nameAccessor,
             saving,
+            selectAccessor,
+            allowEdit,
         } = this.props;
 
-        const { addMode, newItemName, page } = this.state;
+        const {
+            addMode,
+            newItemName,
+            page,
+            editing,
+            itemName,
+        } = this.state;
 
         const items = data.map((item, index) => (
             <Menu.Item
                 key={index.toString()}
                 name={item[nameAccessor]}
+                className={`${indexProp === index ? 'selected-blue' : ''}`}
                 active={indexProp === index}
                 onClick={() => this.handleClickMenuItem(index)}
                 link={indexProp !== index}
             >
-                <span>{item[nameAccessor]}</span>
-                {indexProp === index && saving && (
-                    <Loader active size='tiny' />
-                )}
+                { editing !== index ? (
+                    <React.Fragment>
+                        {selectAccessor && (
+                            <Icon
+                                id={`${item[selectAccessor] ? 'selected' : 'not-selected'}`}
+                                name='grid layout'
+                                onClick={e => this.handleToggle(e, item)}
+                            />
+                        )}
+                        {allowEdit && (
+                            <Icon
+                                id='edit-icon'
+                                name='edit'
+                                onClick={() => this.handleEdit(index, item[nameAccessor])}
+                                data-cy='edit-name-icon'
+                            />
+                        )}
+                        <span>{item[nameAccessor]}</span>
+                        {indexProp === index && saving && (
+                            <Loader active size='tiny' />
+                        )}
+                    </React.Fragment>
+                ) : (
+                    <Input
+                        onChange={this.handleChangeOldName}
+                        value={itemName}
+                        onKeyDown={e => this.handleKeyDownInput(e, item)}
+                        autoFocus
+                        onBlur={this.resetRenameItem}
+                        fluid
+                        data-cy='edit-name'
+                    />
+                )
+                }
+                
             </Menu.Item>
         ));
-
-        if (allowAddition) {
-            items.unshift(
-                !addMode ? (
-                    <Menu.Item
+        return (
+            <React.Fragment>
```

Comment:
You don't have to write React.Fragment, you can just put empty brackets <> </> 
https://reactjs.org/docs/fragments.html#short-syntax